xdsl/transforms/convert_vector_to_ptr.py

```diff
@@ -0,0 +1,68 @@
+from dataclasses import dataclass
+
+from xdsl.context import Context
+from xdsl.dialects import affine, builtin, memref, ptr, vector
+from xdsl.dialects.builtin import (
+    AffineMapAttr,
+    FixedBitwidthType,
+    NoneAttr,
+)
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+
+
+@dataclass
+class VectorLoadToPtr(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: vector.LoadOp, rewriter: PatternRewriter):
+        # Output vector description
+        vector_ty = op.result.type
+        element_type = vector_ty.get_element_type()
+        assert isinstance(element_type, FixedBitwidthType)
+
+        # Input memref description
+        memory = op.base
+        memory_ty = memory.type
+        assert isinstance(memory_ty, memref.MemRefType)
+
+        # Build an affine.apply to compute the linearized offset
+        layout_map = memory_ty.layout
+        if isinstance(layout_map, NoneAttr):
+            layout_map = AffineMapAttr(affine.AffineMap.identity(len(op.indices)))
+        assert isinstance(layout_map, AffineMapAttr)
+        apply_op = affine.ApplyOp(
+            map_operands=op.indices,
+            affine_map=layout_map,
+        )
```

Comment:
We actually have a helper for this:
```suggestion
        layout_map = memory_ty.get_affine_map_in_bytes()
        apply_op = affine.ApplyOp(
            map_operands=op.indices,
            affine_map=AffineMapAttr(layout_map),
        )
```
It would be good to add a test for multiple indices, I don't think that the current code that creates an identity map is correct for multiple dimensions.

Reply:
Done